.github/workflows/todo_app_ci_pipeline.yml

```diff
@@ -40,6 +39,25 @@ jobs:
           password: ${{ secrets.DOCKERHUB_TOKEN }}
       - run: docker build --target production --tag jaisalpatel836/todo_app:latest --tag jaisalpatel836/todo_app:"$GITHUB_SHA" .
       - run: docker push jaisalpatel836/todo_app --all-tags
+      
+  deploy:
+    name: Deploy to Azure via Terraform      
+    needs: publish
+    runs-on: ubuntu-latest
+    steps: 
+      - uses: actions/checkout@v2
+      - name: Run terraform
+        env:
+          ARM_CLIENT_ID: ${{ secrets.TF_VAR_ARM_CLIENT_ID }}
+          ARM_TENANT_ID: ${{ secrets.TF_VAR_ARM_TENANT_ID }}
+          ARM_SUBSCRIPTION_ID: ${{ secrets.TF_VAR_ARM_SUBSCRIPTION_ID }}
+          ARM_CLIENT_SECRET: ${{ secrets.TF_VAR_ARM_CLIENT_SECRET }}
+          ARM_ACCESS_KEY: ${{ secrets.TF_VAR_ARM_ACCESS_KEY }}
```

Comment:
This is a fine solution here but just in case you compare with others you might notice that they don't need to specify this ARM_ACCESS_KEY variable - this is because it looks like you've put your terraform state in a dedicated resource group, and I'm guessing you didn't give the service principal access to that resource group. A very valid way of doing it, I'm just commenting for context!